### ISSUE TITLE
vo=gpu-next support

### ISSUE THREAD

**Issue by Erromango** [jellyfin/jellyfin-media-player#362]
Since latest mpv supports Dolby vision using vo=gpu-next, will Jmp support that?

I’ve tried it with jellyfin shim and it can render dovi video correctly 
But put the same mpv.conf in config folder of jmp it didn’t work.replacing mpv.DLL with latest one also didn’t help.(which works for Plexhtpc on windows)
Jmp log shows on startup vo is set to gpu-next but then be changed to vo=libmpv

**Comment by nyanmisaka** [comment]
I think the next release of JMP should be built with FFmpeg 5.0+ and the latest libplacebo to enable DOVI tone-mapping support. MPV/libmpv have changed their API so you can't just replace the DLL.

**Comment by iwalton3** [comment]
Let me know if updating MPV to version 20230205 e439ddc in the next release fixes this.

**Comment by nyanmisaka** [comment]
It doesn't help at least for me. JMP will always overwrite the `vo` option to `libmpv` instead of the configured `gpu-next`, which is the libplacebo renderer used for DV tone-mapping.

```
2023-02-13 19:29:13 [ DEBUG ] PlayerComponent.cpp @ 588 - cplayer: Setting option 'vo' = 'gpu-next' (flags = 4) 
......
......
2023-02-13 19:29:13 [ DEBUG ] PlayerComponent.cpp @ 588 - cplayer: Set property: vo="libmpv" -> 1 
```

BTW on Linux (debian/ubuntu) JMP is linked against the libmpv/libplacebo/FFmpeg libs from distro. I doubt on some old distros these libs are too old to enable `vo=gpu-next` support, such as on Ubuntu 22.04 (jammy) the FFmpeg version is [4.4.2](https://packages.ubuntu.com/jammy-updates/ffmpeg), which doesn't support DV metadata.

**Comment by nyanmisaka** [comment]
Well, it seems libmpv doesn't support libplacebo (gpu-next) in its yet.
https://github.com/mpv-player/mpv/issues/10810
https://github.com/mpv-player/mpv/issues/6575
https://github.com/mpv-player/mpv/issues/11031

**Comment by iwalton3** [comment]
Yes JMP is using the rendering api. Details are here: https://github.com/jellyfin/jellyfin-media-player/blob/193cacf817dd1c0ac289cb49f61a81c77c2c9fbb/src/player/PlayerQuickItem.cpp

Unfortunately right now the only way to avoid this and allow all of the MPV output options is to use MPV Shim, since all of the UI for it is drawn inside of a window rendered by MPV.

It may be possible to use window embedding which could avoid some of this overhead and support gpu-next now, but it requires different code per platform and may have other issues. https://mpv.io/manual/master/#options-wid

**Comment by RononDex** [comment]
Has there been any update on `gpu-next` support? Noticed that tone-mapping in JellyfinMediaPlayer is off

**Comment by gnattu** [comment]
> Can you please tell us what is the status of the gpu-next support?

It is impossible to use `gpu-next` because upstream mpv maintainers have no interest in implementing it for libmpv renderer api as described in the upstream issues linked earlier. We have to literally start from scratch if we ever want this work.

**Comment by dafyk** [comment]
Jellyfin for AndroidTV has an option to use external player, would be nice to have that feat in desktop client. You wouldnt havve to rewrite everything from scratch and users could just use external mpv with gpu-next support and watch DV content.

**Comment by andrewrabert** [comment]
Just a heads up - development of the Qt based Jellyfin Desktop is effectively stopped in favor of promoting a SDL+CEF rewrite to be the official desktop app. Relevant to this issue specifically, the rewrite uses a fork libmpv with support for gpu-next, vulkan, and HDR. HDR playback is currently implemented and usable on Linux, macOS, and Windows.

It's still early into development, but it is usable. Works great for me as a daily driver on Linux; I welcome any bug reports or feature requests

Not sure on the timeline that this would happen. The more testers we can find, the sooner it'll get a stable release.

The rewrite can be found at https://github.com/jellyfin-labs/jellyfin-desktop-cef

**Comment by BoofOof32** [comment]
I'm happy to see a rewrite for the desktop client, but I'm a little disappointed that Claude is being used to get there. 😕

**Comment by andrewrabert** [comment]
This Qt desktop client has been deprecated in favor of a rewrite: https://github.com/jellyfin/jellyfin-desktop
